pkg/gvk/gvk.go

```diff
@@ -0,0 +1,27 @@
+package gvk
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+type GVKTranslator struct {
+	m *restmapper.DeferredDiscoveryRESTMapper
+}
+
+func NewGVKTranslator(cfg *rest.Config) *GVKTranslator {
+	return &GVKTranslator{m: restmapper.NewDeferredDiscoveryRESTMapper(
+		memory.NewMemCacheClient(
+			discovery.NewDiscoveryClientForConfigOrDie(cfg)))}
```

Comment:
Doesn't this assume a logical cluster?

Reply:
Yeah, it does for now. This PR only works against a single lcluster for now.